src/surface.cc

```diff
@@ -137,7 +137,12 @@ img_data_t SurfaceNode::read_png_or_dat(std::string filename) const
 	img_data_t data;
 	std::vector<uint8_t> png;
 	
-	lodepng::load_file(png, filename);
+	int ret_val = lodepng::load_file(png, filename);
+	
+	if(ret_val == 78){
```

Comment:
78 is just one of the possible return values. Should this be `!= 0` instead?

Reply:
Quoting **Image2** and **Image3** from the previous comment. When the file is not getting successfully open. the **lodepng_filesize()** (Image3 line no 347) returns -1 to load_file() (Image2 line no 6640) and then this function is checking whether the value(size) is less than 0 and returning 78 to the variable ret_val (Image1). 

The reason i kept it 78 as it was already in the code and changing it might affect the code base. as the function might have been called somewhere else as well.

Reply:
Please don't post images of source code as references.  You can create links directly to the source on github as a better alternative.
For example :
1) go to: https://github.com/openscad/openscad
2) Press 't' hotkey, to search by filename
3) type 'lodepng' and click on the cpp file to view
4) a) For a single line you can just click the line number (374) for example
    b) To select multiple lines click the first line number in the left column (368), then shift-click the end line (381) 
5) Click the "..." that should now show on the left of your selection
6) "Copy Permalink", then paste that into a comment.  It will show up as follows:

https://github.com/openscad/openscad/blob/2a697e4090cfdb0e376dd5565eaad104dde0c133/src/ext/lodepng/lodepng.cpp#L368-L381

Reply:
ok, I see what's going on now; hard to read from the screenshots:
* If the file cannot be opened or read, lodepng::load_file() will return 78
* If the file buffer cannot be allocated lodepng::load_file() will throw std::bad_alloc

You're only handling the first case, which is fine for now, but you error message should reflect that.

Reply:
will writing a try catch block do to handle the second case as well?

Reply:
Yes, I believe so.

Reply:
I have written a try a try catch block for handling the bad_alloc error.

Reply:
> ok, I see what's going on now; hard to read from the screenshots:
> 
>     * If the file cannot be opened or read, lodepng::load_file() will return 78
> 
>     * If the file buffer cannot be allocated lodepng::load_file() will throw std::bad_alloc
> 
> 
> You're only handling the first case, which is fine for now, but your error message should reflect that.

Should i write a try and catch block?

Reply:
> Yes, I believe so.

I accidently deleted my previous comment is there a way to get it back? I was me asking should i write a try catch block?